File: sorting_algorithms/bubble_sort.c
#include <stdio.h>

// Function to implement bubble sort
void bubbleSort(int arr[], int n) {
    int i, j, temp;
    int swapped;
    
    // Outer loop to iterate over the entire array
    for (i = 0; i < n-1; i++) {
        swapped = 0;
        
        // Inner loop to compare adjacent elements
        for (j = 0; j < n-i-1; j++) {
            // Swap if the element found is greater than the next element
            if (arr[j] > arr[j+1]) {
                temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                swapped = 1;
            }
        }
        
        // If no elements were swapped, break out of the loop
        if (swapped == 0)
            break;
    }
}

// Function to print an array
void printArray(int arr[], int size) {
    int i;
    for (i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

// Main function to test the bubble sort algorithm
int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr)/sizeof(arr[0]);
    
    printf("Unsorted array: \n");
    printArray(arr, n);
    
    bubbleSort(arr, n);
    
    printf("Sorted array: \n");
    printArray(arr, n);
    
    return 0;
}

File: sorting_algorithms/bubble_sort.js
function bubbleSort(arr) {
    let n = arr.length;
    let swapped;
    
    // Outer loop to iterate over the entire array
    for (let i = 0; i < n; i++) {
        swapped = false;
        
        // Inner loop to compare adjacent elements
        for (let j = 0; j < n - i - 1; j++) {
            // Swap if the element found is greater than the next element
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swapped = true;
            }
        }
        
        // If no elements were swapped, break out of the loop
        if (!swapped) {
            break;
        }
    }

    console.log(arr);
}

let arr = [5, 9, 10, 2, 23, 1, 4 ]

bubbleSort(arr)